go/mcap/reader_options.go

```diff
@@ -4,6 +4,19 @@ import (
 	"fmt"
 )
 
+type ReaderOptions struct {
+	LexerOpts *LexerOptions
+}
+
+type ReaderOpt func(*ReaderOptions) error
+
+func WithLexerOptions(lexerOpts *LexerOptions) ReaderOpt {
+	return func(ro *ReaderOptions) error {
+		ro.LexerOpts = lexerOpts
+		return nil
+	}
+}
```

Comment:
I'm not totally sure this is the best config option to publicly expose. The reader uses a lexer in two places I believe. This config is for one of them. But I'm not sure that detail is good for us to expose to the user. If we need to add an option, something explicit like "AttachmentCallback" seems better/more robust to change IMO.

I'm not totally sold that putting this functionality into the Reader is the way to go. I think we could even get by without reader changes, if in the CLI we iterate over the attachment indexes in each Reader (obtained with the Info) call, and then use the `reader.GetAttachment` function to access them and stream them into the output writer.

Reply:
> iterate over the attachment indexes

I like this idea, but I assume this requires an indexed mcap (which `merge` does not today)?  Same problem if I set up an attachment callback (similar to metadataCallback) in ReadOpt, at least with the current lexer implementation. Currently, the only hook into OpAttachment is the lexer's callback.

I pushed a commit that explores this; maybe we can discuss based on that.

Reply:
Discussed offline: a reasonable compromise is to make a second pass over the file just using a lexer, and let its attachment callback process the attachments.